_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,416 @@
+This requires a tool that can assign the partitions from the broker to be removed, to the remaining brokers.
+The most convenient tool for this job is the Kafka partition reassignment tool.
+
+<!--more-->
+
+##  Kafka reassignment partition tool
+
+The `bin/kafka-reassign-partitions.sh` tool allows you to reassign partitions to different brokers.
+
+While using the tool, you have to provide it with either of these two JSON files: `topics.json` and `reassignment.json`.
+Wondering what these two JSON files really do? Let's talk a bit about them:
+
+- The `topic.json` basically consists of the topics that we want to reassign or move. Based on this JSON file, our tool will generate a proposal `reassignment.json` that we can use directly or modify further.
+
+- The `reassignment.json` file is a configuration file that is used during the partition reassignment process. The reassignment partition tool will generate a proposal `reassignment.json` file based on a `topics.json` file. You can change the `reassignment.json` file as per your requirement and use it.
+
+## Why use the Kafka reassignment partition tool?
+
+The Kafka reassignment partition tool can help you address a variety of use cases.
+
+Some of these are listed here:
+
+1. You can reassign the partition between the brokers anytime. For example, it can be used at times when you want to scale down the number of brokers in your cluster.
+   You can assign partitions from the broker to be scaled down to other brokers which will handle these partitions now.
+
+2. With the help of this tool, you can increase the number of partitions / replicas which can help with increasing the throughput of the topic.
+
+## Partition Reassignment Throttle
+Reassigning partitions between brokers can sometimes lead to a large transfer of large data.
+To avoid overloading the cluster, it is recommended to always set a throttle rate to limit the bandwidth used by the reassignment. 
+This can be done using the `--throttle` flag which sets the maximum allowed bandwidth in bytes, for example `--throttle 5000000` sets the limit to 50 MB/s.
+
+Throttling might cause the reassignment to take longer to complete.
+
+1. If the throttle is too low, the newly assigned brokers will not be able to keep up with records being published and the reassignment will never complete.
+
+2. If the throttle is too high, the overall health of the cluster may be impacted.
+
+## Actions that can be executed while using the tool
+
+It has three different actions:
+
+| Action | Description |
+| :-----------: | ------------- |
+| `generate`  | Takes a set of topics and brokers and generates a reassignment JSON file. This actions is optional if we already have `reassignment.json` file with us.|
+| `execute` | Takes a reassignment JSON file and applies it to the partitions and brokers in the cluster. The `reassignment.json` file can be either the one proposed by the `--generate` command or written by the user itself. |
```

Comment:
> `--generate` command

You just referred to it as the "generate action". You should try to use consistent terminology. So you should probably explain that the tool has these modes (which you have), _which are selected using the correspondingly named option_ (which you're missing).